source/solvers/vof_assemblers.cc

```diff
@@ -227,8 +240,10 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
           const auto grad_phi_phase_i = scratch_data.grad_phi[q][i];
 
 
-          // rhs for : u * nabla(phase) = 0
-          local_rhs(i) -= (phi_phase_i * velocity * phase_gradient) * JxW;
+          // rhs for: u * grad(phase) - diffusivity * laplacian(phase) = 0
+          local_rhs(i) -= (phi_phase_i * velocity * phase_gradient +
+                           diffusivity * grad_phi_phase_i * phase_gradient) *
```

Comment:
These added terms (to the local matrix, local rhs, strong residual, ...) should be added using a new assembler (not in the VOFAssemblerCore).

Reply:
Most of the calculation for peeling/wetting mechanism is done in the vof.cc.
Here `diffusivity` is 0 by default, and when it is so, the assembler is the same as before. Also, it is not mandatory to have a diffusivity for the wetting mechanism, it is only a mean to improve it in certain cases, as explained in the documentation. 

So I think putting everything in another assembler would be extra verbose/code doubling and I do not really see the gain from it. For what I understand you are asking for, I would have to add like a `VOFAssemblerCoreSkipConservation` and `VOFAssemblerCoreNonNewtonianSkipConservation` which would be 98% copy/paste from the `VOFAssemblerCore` and `VOFAssemblerCoreNonNewtonian`. Did I understand your request correctly?

In my opinion, code doubling would be more of a problem, but I can still do it if you think it is better.

Reply:
yes, I mean using a new assembler to add the new terms, but if you think these new terms are related to VOFAssemblerCore (and not to the wetting/peeling), that's OK with me.

Reply:
Well the "skip mass conservation" part is related to the VOF in general, it can improve wetting but it can be used even when peeling/wetting are not enabled (I will add a part to the break dam example about it). As for the diffusivity part, you are right that it is only used to improve the wetting mechanism. If you prefer it to be more separated, maybe I can add like a VOFAssembleDiffusivity, called after the VOFAssembleCore when `diffusivity` is not zero? It would involve relooping again on all the dof, but it would certainly be clearer